components/email_aliases/browser/email_aliases_service_factory.h

```diff
@@ -0,0 +1,36 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_EMAIL_ALIASES_BROWSER_EMAIL_ALIASES_SERVICE_FACTORY_H_
+#define BRAVE_COMPONENTS_EMAIL_ALIASES_BROWSER_EMAIL_ALIASES_SERVICE_FACTORY_H_
+
+#include <memory>
+
+#include "base/memory/singleton.h"
+#include "components/keyed_service/content/browser_context_keyed_service_factory.h"
+
+namespace email_aliases {
+
+class EmailAliasesService;
+
+class EmailAliasesServiceFactory : public BrowserContextKeyedServiceFactory {
+ public:
+  static EmailAliasesService* GetForProfile(Profile* profile);
```

Comment:
it's actually better that we don't expose the service as a raw pointer and instead provide a factory method that returns mojom::EmailAliasesService. Something like `mojo::PendingRemote<mojom::EmailAliasesService> GetForProfile(...)`

`

Reply:
`GetForProfile` is being used here: https://github.com/brave/brave-core/pull/29700/files#diff-38b6c12d746e933cd876120ea21df0a3ddc37a4879dbed20f364640ad3c60bdeR268. I think I need a pointer to the instance to bind the interface, right? Or should I be doing this in a different way?

Reply:
Do it inside the method I described above so you never expose the raw pointer

Reply:
I have added a factory method `BindForProfile(...)`